File: cypress/sauce-demo-automation/cypress/support/pages/CheckoutPage.js
class CheckoutPage {
    get firstName() {
        return cy.get('#first-name');
    }

    get lastName() {
        return cy.get('#last-name');
    }

    get postalCode() {
        return cy.get('#postal-code');
    }

    get continueButton() {
        return cy.get('#continue');
    }

    get finishButton() {
        return cy.get('#finish');
    }

    get confirmationMessage() {
        return cy.get('.complete-header');
    }

    get errorMessage() {
        return cy.get('.error-message-container');
    }

    enterShippingInformation(fname, lname, postal) {
        this.firstName.type(fname);
        this.lastName.type(lname);
        this.postalCode.type(postal);
    }

    clickContinue() {
        this.continueButton.click();
    }

    clickFinish() {
        this.finishButton.click();
    }

    getConfirmationMessage() {
        return this.confirmationMessage.invoke('text');
    }

    isErrorMessageDisplayed() {
        return this.errorMessage.should('be.visible');
    }
}

export default new CheckoutPage();

File: java-selenium/sauce-demo-automation/src/test/java/com/saucedemo/automation/pages/CheckoutPage.java
package com.saucedemo.automation.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class CheckoutPage {
    private WebDriver driver;

    @FindBy(id = "first-name")
    private WebElement firstName;

    @FindBy(id = "last-name")
    private WebElement lastName;

    @FindBy(id = "postal-code")
    private WebElement postalCode;

    @FindBy(id = "continue")
    private WebElement continueButton;

    @FindBy(id = "finish")
    private WebElement finishButton;

    @FindBy(css = ".complete-header")
    private WebElement confirmationMessage;

    @FindBy(css = ".error-message-container")
    private WebElement errorMessage;

    public CheckoutPage(WebDriver driver) {
        this.driver = driver;
        PageFactory.initElements(driver, this);
    }

    public void enterShippingInformation(String fname, String lname, String postal) {
        firstName.sendKeys(fname);
        lastName.sendKeys(lname);
        postalCode.sendKeys(postal);
    }

    public void clickContinue() {
        continueButton.click();
    }

    public void clickFinish() {
        finishButton.click();
    }

    public String getConfirmationMessage() {
        return confirmationMessage.getText();
    }

    public boolean isErrorMessageDisplayed() {
        return errorMessage.isDisplayed();
    }

    public String getCurrentUrl() {
        return driver.getCurrentUrl();
    }
}
